components/03-organisms/galleries/media-grid/_yds-media-grid-modal.scss

```diff
@@ -1,27 +1,236 @@
+@use '../../../00-tokens/tokens';
+@use '../../../01-atoms/atoms';
+
+$modal-speed: var(--animation-speed-slow);
+
+@mixin media-grid-modal-transition {
+  @include tokens.animate(opacity, $modal-speed);
+}
+
 .media-grid__modal {
-  display: none;
   position: fixed;
   top: 0;
   left: 0;
   height: 100%;
   width: 100%;
 
+  // This z-index is crazy high to ensure it shows above admin toolbars.
+  z-index: 600;
+  background: transparent;
+
+  [data-media-grid-modal-state='inactive'] & {
+    @include tokens.animate-hidden($modal-speed);
+    @include media-grid-modal-transition;
+
+    opacity: 0;
+  }
+
+  // Show the modal and backdrop when active.
   [data-media-grid-modal-state='active'] & {
-    display: grid;
-    place-content: center;
+    @include media-grid-modal-transition;
+
+    opacity: 1;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &::before {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    background-color: var(--color-background);
+    opacity: 0.9;
+  }
+
+  // Position modal content relative, to "place it on top of the backdrop".
+  > * {
+    position: relative;
+  }
+
+  svg {
+    max-width: 100%;
+    max-height: 100%;
   }
 }
 
-.media-grid__modal-backdrop {
-  position: absolute;
-  top: 0;
-  left: 0;
+.media-grid-modal__item {
+  padding-block: var(--size-spacing-site-gutter);
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+}
+
+.media-grid-modal__inner {
+  display: grid;
+
+  // Stylelint is mis-interpreting the `/` below as a division operator, rather
+  // than as a part of the grid-template sizing stuff. So, we need to disable
+  // that check for this chunk of code.
+  /* stylelint-disable scss/operator-no-newline-before */
```

Comment:
This seems fine to disable stylint here. The way you have `grid-template` formatted here is much easier to read, :tada: